utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -653,6 +665,67 @@ int main(int argc, char **argv)
 
                     break;
                 }
+                case 32:
+                {
+                    testCaseName = "erase";
+
+                    Rpp8u *colors8u = reinterpret_cast<Rpp8u *>(colorBuffer);
+                    Rpp16f *colors16f = reinterpret_cast<Rpp16f *>(colorBuffer);
+                    Rpp32f *colors32f = colorBuffer;
+                    Rpp8s *colors8s = reinterpret_cast<Rpp8s *>(colorBuffer);
+                    int idx;
+
+                    init_erase(batchSize, boxesInEachImage, numOfBoxes, anchorBoxInfoTensor, roiTensorPtrSrc, srcDescPtr->c, colorBuffer);
+
+                    for(int i = 0; i < batchSize; i++)
+                    {
+                        if(srcDescPtr->c == 3)
+                        {
+                            idx = boxesInEachImage * 3 * i;
+                            for (int j = 0; j < 9; j++)
+                            {
+                                if (!inputBitDepth)
+                                    colors8u[idx + j] = (Rpp8u)(colorBuffer[idx + j]);
```

Comment:
it is better to have a templated function to assign this based on inputBitDepth. 
I suggest to move all common functions in a separate file for all the unit_tests so there is less repeated code in each case

Reply:
Moved to separate file, and common code has been shrunk